halo2_proofs/src/plonk/prover.rs

```diff
@@ -473,6 +497,17 @@ where
         })
         .collect::<Result<Vec<_>, _>>()?;
 
+    let shuffles: Vec<Vec<shuffle::prover::Committed<Scheme::Curve>>> = shuffles
+        .into_iter()
+        .map(|shuffles| -> Result<Vec<_>, _> {
+            // Construct and commit to products for each shuffle
+            shuffles
+                .into_iter()
+                .map(|shuffle| shuffle.commit_product(pk, params, gamma, &mut rng, transcript))
+                .collect::<Result<Vec<_>, _>>()
+        })
+        .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Merged with `compress`
```suggestion
    let shuffles: Vec<Vec<shuffle::prover::Committed<Scheme::Curve>>> = instance
        .iter()
        .zip(advice.iter())
        .map(|(instance, advice)| -> Result<Vec<_>, _> {
            pk.vk
                .cs
                .shuffles
                .iter()
                .map(|shuffle| {
                    // Compress expressions for each shuffle, then construct and commit to products for each shuffle
                    shuffle
                        .compress(
                            pk,
                            params,
                            domain,
                            theta,
                            &advice.advice_polys,
                            &pk.fixed_values,
                            &instance.instance_values,
                            &challenges,
                        )
                        .commit_product(pk, params, gamma, &mut rng, transcript)
                })
                .collect::<Result<Vec<_>, _>>()
        })
        .collect::<Result<Vec<_>, _>>()?;
```